test_fms/diag_integral/test_diag_integral.F90

```diff
@@ -96,6 +96,24 @@ program test_diag_integral
   call test_sum_diag_integral_field !< compare read in values to the expected values.
 
 contains
+  !-------------------------------------
+  !-------------------------------------
+  subroutine initialize_arrays
+
+    !> made up numbers
+
+    implicit none
+
+    lon=1.0_lkind
+    lat=1.0_lkind
+    area=1.0_lkind
+    immadeup2=1.0_lkind
+    immadeup3=1.0_lkind
+    immadeupw=1.0_lkind
+    immadeuph=1.0_lkind
+    weight=1.0_lkind
+
+  end subroutine initialize_arrays
```

Comment:
Why not get rid of `initialize_arrays` and initialize these in the declaration statements at the top of the program instead? As a bonus, that would give them an implicit `save` attribute.